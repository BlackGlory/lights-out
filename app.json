{
  "expo": {
    "name": "Lights Out",
    "slug": "lights-out",
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#FFFFFF"
      },
      "package": "me.blackglory.lights_out",
      "permissions": []
    },
    "web": {
      "favicon": "./assets/favicon.png"
    }
  }
}
