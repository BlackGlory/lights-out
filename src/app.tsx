import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createNativeStackNavigator } from '@react-navigation/native-stack'
import { Home } from './screens/home'
import { Level1 } from './screens/level-1'
import { Level2 } from './screens/level-2'
import { Level3 } from './screens/level-3'
import { Level4 } from './screens/level-4'
import { Level5 } from './screens/level-5'

const Stack = createNativeStackNavigator()

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name='Home' component={Home} options={{ headerShown: false }} />
        <Stack.Screen name='Level 1' component={Level1} />
        <Stack.Screen name='Level 2' component={Level2} />
        <Stack.Screen name='Level 3' component={Level3} />
        <Stack.Screen name='Level 4' component={Level4} />
        <Stack.Screen name='Level 5' component={Level5} />
      </Stack.Navigator>
    </NavigationContainer>
  )
}
