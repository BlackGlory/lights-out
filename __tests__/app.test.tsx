import { App } from '../src/app'
import { render } from '@testing-library/react-native'

// fix react-navigation issues:
// - https://github.com/react-navigation/react-navigation/issues/9727
// - https://github.com/react-navigation/react-navigation/issues/9568
jest.useFakeTimers()

it('renders correctly', () => {
  const { toJSON } = render(<App />)

  expect(toJSON()).toMatchSnapshot()
})
