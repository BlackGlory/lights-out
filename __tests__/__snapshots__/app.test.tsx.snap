// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        hideShadow={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "height": "100%",
              "justifyContent": "center",
            }
          }
          contentInsetAdjustmentBehavior="automatic"
          style={
            Object {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginBottom": 32,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 96,
                    "marginBottom": 10,
                    "width": 96,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontFamily": "serif",
                    "fontSize": 40,
                    "fontWeight": "bold",
                  }
                }
              >
                Lights Out
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginVertical": 16,
                    "width": 128,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "textAlign": "center",
                        }
                      }
                    >
                      Level 1
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginVertical": 16,
                    "width": 128,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "textAlign": "center",
                        }
                      }
                    >
                      Level 2
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginVertical": 16,
                    "width": 128,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "textAlign": "center",
                        }
                      }
                    >
                      Level 3
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginVertical": 16,
                    "width": 128,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "textAlign": "center",
                        }
                      }
                    >
                      Level 4
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginVertical": 16,
                    "width": 128,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "black",
                        "padding": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "textAlign": "center",
                        }
                      }
                    >
                      Level 5
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
